python/caffe/test/test_python_layer.py

```diff
@@ -44,6 +44,23 @@ def forward(self, bottom, top):
     def backward(self, top, propagate_down, bottom):
         self.blobs[0].diff[0] = 1
 
+class PhaseLayer(caffe.Layer):
+    """A layer for checking attribute `phase`"""
+
+    def setup(self, bottom, top):
+        pass
+
+    def reshape(self, bootom, top):
+        top[0].reshape(1)
+
+    def forward(self, bottom, top):
+        if (self.phase == caffe.TRAIN):
+            top[0].data[...] = 0 # caffe.TRAIN
```

Comment:
Please never hardcode the values of symbolic constants (note that these really need to be correct, because they're checked later).

Why not replace this whole forward with `top[0].data[()] = self.phase`?